modules/tensor_mechanics/src/materials/ADPowerLawCreepStressUpdate.C

```diff
@@ -34,15 +33,17 @@ template <ComputeStage compute_stage>
 ADPowerLawCreepStressUpdate<compute_stage>::ADPowerLawCreepStressUpdate(
     const InputParameters & parameters)
   : ADRadialReturnCreepStressUpdateBase<compute_stage>(parameters),
-    _has_temp(parameters.isParamSetByUser("temperature")),
-    _temperature(_has_temp ? coupledValue("temperature") : _zero),
+    _has_temp(isParamValid("temperature")),
     _coefficient(adGetParam<Real>("coefficient")),
     _n_exponent(adGetParam<Real>("n_exponent")),
     _m_exponent(adGetParam<Real>("m_exponent")),
     _activation_energy(adGetParam<Real>("activation_energy")),
     _gas_constant(adGetParam<Real>("gas_constant")),
     _start_time(adGetParam<Real>("start_time"))
 {
+  if (_has_temp)
+    _temperature = &adCoupledValue("temperature");
```

Comment:
I would just change that to 

```
_temperature(isParamValid("temperature") ? &adCoupledValue("temperature") : nullptr)
```

in the initializer list and do 

```
if (_temperature)
```

below. Then you can get rid of `_has_temperature` and the code reads just as well if not better.

Reply:
For further micro optimization you can initialize `_exponential(1.0)` in the constructor and drop the `else` clause in `computeStressInitialize`.